Adds missing python-ldap dependency to LDAP extra.

potiuk opened the pull request:
Please check only the last commit as this is based on #13329 

It seems that for quite some time (1.10.4) the "ldap" extra
missed python-ldap dependency.

https://issues.apache.org/jira/browse/AIRFLOW-5261

Also LDAP seems to be popular enough to be added as default
extra in the production image.

Fixes #13306

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

kaxil reviewed (dismissed):
Looks like `kerberos` and `statsd` are also missing, can you add them too in this PR and rename it too "Adds missing extra dependencies" ?

kaxil commented on setup.py:
```diff
@@ -766,9 +766,9 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     'jenkins': ["jenkins"],
     'jira': ["jira"],
-    'kerberos': [],
+    'kerberos': ["kerberos"],
```
Jarek, looks like `apache.beam` one is also empty (Line 718) -- should it be empty, it should have `apache.beam` right?

Similarly I am not sure if the following should be empty too:

- dask
- apache.atlas
- rabbitmq
- sentry
- tableau

WDYT ?

potiuk replied:
I will review it fully,

potiuk replied:
actually I remove back statsd/kerberos. We do not have neither of them as providers :)

Similarly there are no providers for dask, apache.atlas, rabbitmq, sentry. There is one for tableau but it correctly has no additional requirements. 

So just correcting LDAP was right intiially. I removed stasd/kerberos.

potiuk replied:
In our case extras <> providers. We have  quite a few more extras than providers . We need to refactor setup.py badly  to reflect that because it is awfully confusing (I was confused myself as you see) . We already discussed it in #12548 and I am going to do this very thing before New Year. I created #13309 to not forget about it.

potiuk replied:
🤦 🤦 . There is no `ldap` provider either :(.  I thougght about IMAP when I thought it is missing. This is a different issue altogether

potiuk replied:
Seems that we simply miss `python-ldap` dependency (and it goes as far as 1.10.4 https://issues.apache.org/jira/browse/AIRFLOW-5261 and  https://apache-airflow.slack.com/archives/C0146STM600/p1608836687492700

potiuk committed:
Adds missing LDAP "extra" dependencies to ldap provider.

It seems that for quite some time (1.10.4) the "ldap" extra
missed python-ldap dependency.

https://issues.apache.org/jira/browse/AIRFLOW-5261

Also LDAP seems to be popular enough to be added as default
extra in the production image.

Fixes #13306
